File: netapp-manageability-sdk-9.3/src/sample/Core_API/Ruby/hello_ontapi.rb
#============================================================#
#                                                            #
# $ID:$                                                      #
#                                                            #
# hello_ontapi.rb                                            #
#                                                            #
# "Hello_world" program which prints the ONTAP version       #
# number of the destination storage system                   #
#                                                            #
# Copyright 2011 Network Appliance, Inc. All rights          #
# reserved. Specifications subject to change without notice. #
#                                                            #
# This SDK sample code is provided AS IS, with no support or #
# warranties of any kind, including but not limited to       #
# warranties of merchantability or fitness of any kind,      #
# expressed or implied.  This code is subject to the license #
# agreement that accompanies the SDK.                        #
#                                                            #
# tab size = 8                                               #
#                                                            #
#============================================================#

$:.unshift '../../../../lib/ruby/NetApp'
require 'NaServer'

def print_usage
    print ("Usage: hello_ontapi.rb <storage_system> <user> <password> \n")
    print ("<storage> -- storage_system\n")
    print ("<user> -- User name\n")
    print ("<password> -- Password\n")
    exit
end
	

args = ARGV.length
if(args < 3)
    print_usage
end
storage = ARGV[0]
user = ARGV[1]
password = ARGV[2]

s = NaServer.new(storage, 1, 1)
s.set_server_type("Filer")
s.set_admin_user(user, password)
s.set_transport_type("HTTP")
output = s.invoke("system-get-version")

if(output.results_errno() != 0)
    r = output.results_reason()
    print("Failed : \n" + r)
else 
    r = output.child_get_string("version")
    print ("Hello World ! DOT version is : " + r + "\n")
end

File: netapp-manageability-sdk-9.3/src/sample/Core_API/Perl_Bindings/hello_ontapi.pl
#============================================================#
#                                                            #
# $ID:$                                                      #
#                                                            #
# hello_ontapi.pl                                            #
#                                                            #
# "Hello_world" program which prints the ONTAP version       #
# number of the destination Storage                          #
#                                                            #
# Copyright 2013 Network Appliance, Inc. All rights          #
# reserved. Specifications subject to change without notice. #
#                                                            #
# This SDK sample code is provided AS IS, with no support or #
# warranties of any kind, including but not limited to       #
# warranties of merchantability or fitness of any kind,      #
# expressed or implied.  This code is subject to the license #
# agreement that accompanies the SDK.                        #
#                                                            #
# tab size = 8                                               #
#                                                            #
#============================================================#

my $VERSION = '9.3';	# Controls the SDK release.
use strict;

use lib '../../../../lib/perl/NetApp';
use NaServer;
use Pod::Usage;

die pod2usage( 
	verbose => 1
) unless $ARGV[2];

our $server = $ARGV[0];
our $user = $ARGV[1];
our $password = $ARGV[2];
our $output ;

our $s = NaServer->new($server, 1, 1);
$s->set_admin_user($user, $password);

eval {
    $output = $s->system_get_version();
};

if($@) {
    print "Failed : $@";
}
else {
    our $version = $output->{'version'};
    print "$version\n";
}
#=========================== POD ============================#

=head1 NAME

  hello_ontapi.pl - Basic ONTAPI Test

=head1 SYNOPSIS

  hello_ontapi.pl <server name> <user> <password>

=head1 ARGUMENTS

  <server Name>
  The name of the Storage Server to test.

  <user>
  username. 

  <password>
  password.

=head1 SEE ALSO

  NaServer.pm, OntapClusterAPI.pm, Ontap7ModeAPI.pm

=head1 COPYRIGHT

  Copyright 2013 Network Appliance, Inc. All rights 
  reserved. Specifications subject to change without notice.

  This SDK sample code is provided AS IS, with no support or 
  warranties of any kind, including but not limited to 
  warranties of merchantability or fitness of any kind, 
  expressed or implied.  This code is subject to the license 
  agreement that accompanies the SDK.

=cut





